internal/cmd/deployrequest/deploy.go

```diff
@@ -47,15 +54,64 @@ func DeployCmd(ch *cmdutil.Helper) *cobra.Command {
 				}
 			}
 
-			if ch.Printer.Format() == printer.Human {
-				ch.Printer.Printf("Successfully queued %s from %s for deployment to %s.\n",
-					dr.ID, dr.Branch, dr.IntoBranch)
-				return nil
+			// wait and check until the deploy request is deployed
+			if flags.wait {
+				end := ch.Printer.PrintProgress(fmt.Sprintf("Waiting until deploy request %s/%s is deployed...",
+					printer.BoldBlue(database), printer.BoldBlue(number)))
+				defer end()
+				getReq := &planetscale.GetDeployRequestRequest{
+					Organization: ch.Config.Organization,
+					Database:     database,
+					Number:       n,
+				}
+				if err := waitUntilReady(ctx, client, ch.Printer, ch.Debug(), getReq); err != nil {
+					return err
+				}
+				end()
+
+				ch.Printer.Printf("Deploy request %s/%s is successfully deployed.",
+					printer.BoldBlue(database), printer.BoldBlue(number))
+
+			} else {
+				if ch.Printer.Format() == printer.Human {
+					ch.Printer.Printf("Successfully queued %s from %s for deployment to %s.\n",
+						dr.ID, dr.Branch, dr.IntoBranch)
+					return nil
+				}
 			}
 
 			return ch.Printer.PrintResource(toDeployRequest(dr))
 		},
 	}
 
+	cmd.Flags().BoolVar(&flags.wait, "wait", false, "wait until the branch is ready")
```

Comment:
🤔 should this say `wait until the branch is deployed`
```suggestion
	cmd.Flags().BoolVar(&flags.wait, "wait", false, "wait until the branch is deployed")
```